quaternions.scad

```diff
@@ -327,4 +447,216 @@ function Qrot(q,p) =
 module Qrot_copies(quats) for (q=quats) Qrot(q) children();
 
 
+// Function: Q_Rotation()
+// Usage:
+//   Q_Rotation(R)
+// Description:
+//   Returns a normalized quaternion corresponding to the rotation matrix R.
+//   R may be a 3x3 rotation matrix or a homogeneous 4x4 rotation matrix.
+//   The last row and last column of R are ignored for 4x4 matrices.
+//   It doesn't check whether R is in fact a rotation matrix.
+//   If R is not a rotation, the returned quaternion is an unpredictable quaternion .
+function Q_Rotation(R) =
+  assert( is_matrix(R,3,3) || is_matrix(R,4,4) , 
+                    "Matrix is neither 3x3 nor 4x4")
+  let( tr = R[0][0]+R[1][1]+R[2][2] ) // R trace
+  tr>0
+  ? let( r = 1+tr  )
+    _Qnorm( _Qset([ R[1][2]-R[2][1], R[2][0]-R[0][2], R[0][1]-R[1][0] ], -r ) )
+  : let( i = max_index([ R[0][0], R[1][1], R[2][2] ]),
+         r = 1 + 2*R[i][i] -R[0][0] -R[1][1] -R[2][2] )
+    i==0 ? _Qnorm( _Qset( [ 4*r, (R[1][0]+R[0][1]), (R[0][2]+R[2][0]) ], (R[2][1]-R[1][2])) ):
+    i==1 ? _Qnorm( _Qset( [ (R[1][0]+R[0][1]), 4*r, (R[2][1]+R[1][2]) ], (R[0][2]-R[2][0])) )
+         : _Qnorm( _Qset( [ (R[2][0]+R[0][2]), (R[1][2]+R[2][1]), 4*r ], (R[1][0]-R[0][1])) );
+
+
+// Function&Module: Q_Rotation_path(q1, n, [q2])
+// Usage as a function:

```

Comment:
Usage line syntax, to be properly parsed by the documents generator, should have the `:` at the end of the `Usage:` word.  Everything following that is the usage type title.
```
// Usage: As a module
//   path = Q_Rotation_path(q1, n, q2);

Reply:
Noted.